retrieval/k8s/k8s_stats.go

```diff
@@ -194,36 +245,37 @@ func sanitizeData(to interface{}) interface{} {
 		cast.Spec.AllocateLoadBalancerNodePorts = nil
 		cast.Spec.LoadBalancerClass = nil
 		cast.Spec.InternalTrafficPolicy = nil
-		return cast
+		return cast, nil
 	case *corev1.ReplicationController:
 		cast := to.(*corev1.ReplicationController)
 		sanitizeMeta(&cast.ObjectMeta)
 		cast.Spec.Replicas = nil
 		cast.Spec.Template = nil
 		cast.Spec.MinReadySeconds = 0
-		return cast
+		return cast, nil
 	case *corev1.Namespace:
-		return sanitizeNamespace(to)
+		return sanitizeNamespace(to), nil
 	case *corev1.PersistentVolume:
 		cast := to.(*corev1.PersistentVolume)
 		sanitizeMeta(&cast.ObjectMeta)
-		return cast
+		return cast, nil
 	case *corev1.PersistentVolumeClaim:
 		cast := to.(*corev1.PersistentVolumeClaim)
 		sanitizeMeta(&cast.ObjectMeta)
-		return cast
+		return cast, nil
 	case *corev1.Node:
 		cast := to.(*corev1.Node)
 		sanitizeMeta(&cast.ObjectMeta)
-		return cast
+		return cast, nil
 	}
-	return to
+	return to, fmt.Errorf("invalid object type")
```

Comment:
Any particular reason we want to surface an error from this?

It shouldn't occur as we control both the informers watched and this list of object types. Seems like we'd be introducing an unnecessary risk for minimal value.

Reply:
I am okay with dropping this, it just seemed a little strange swallowing the error, but it should be fine. If it is none of the types it will just drop the resource anyway